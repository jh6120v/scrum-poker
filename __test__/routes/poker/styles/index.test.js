import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import {
    PokerContainer, PokerItem, PokerItemInner, PokerItemMask,
    PokerItemWrap,
    PokerListWrap, PokerSelectContainer, PokerSelectedItemFront, PokerSelectMask, PokerSelectSafari,
    PokerSelectWrap,
    PokerWrap
} from '../../../../src/routes/poker/styles';

describe('test Poker style.', () => {
    it('test <PokerWrap /> snapshot.', () => {
        const { container } = render(<PokerWrap />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerWrap /> when render it.', () => {
        const { container } = render(<PokerWrap />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', '100%');
        expect(tree).toHaveStyleRule('background-color', '#aadff0');
        expect(tree).toHaveStyleRule('padding-top', '44px');
    });

    it('test <PokerContainer /> snapshot.', () => {
        const { container } = render(<PokerContainer />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerContainer /> when render it.', () => {
        const { container } = render(<PokerContainer />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '350px', {
            media: '(min-width: 768px)'
        });
        expect(tree).toHaveStyleRule('margin', '0 auto', {
            media: '(min-width: 768px)'
        });
    });

    it('test <PokerListWrap /> snapshot.', () => {
        const { container } = render(<PokerListWrap />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerListWrap /> when render it.', () => {
        const { container } = render(<PokerListWrap />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-wrap', 'wrap');
        expect(tree).toHaveStyleRule('justify-content', 'space-between');
        expect(tree).toHaveStyleRule('align-content', 'center');
        expect(tree).toHaveStyleRule('z-index', '1');
        expect(tree).toHaveStyleRule('padding', '0 20px');
    });

    it('test <PokerSelectWrap /> snapshot.', () => {
        const { container } = render(<PokerSelectWrap />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerSelectWrap /> when render it.', () => {
        const { container } = render(<PokerSelectWrap />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('z-index', '1');
        expect(tree).toHaveStyleRule('padding', '0 20px');
    });

    it('test <PokerItemWrap /> snapshot.', () => {
        const { container } = render(<PokerItemWrap />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerItemWrap /> when render it.', () => {
        const { container } = render(<PokerItemWrap />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('width', '100px');
        expect(tree).toHaveStyleRule('max-width', '22%');
        expect(tree).toHaveStyleRule('margin', '6.5px 0');
        expect(tree).toHaveStyleRule('content', "''", {
            modifier: '::after'
        });
        expect(tree).toHaveStyleRule('display', 'block', {
            modifier: '::after'
        });
        expect(tree).toHaveStyleRule('padding-bottom', '140%', {
            modifier: '::after'
        });
    });

    it('test <PokerItem /> snapshot.', () => {
        const { container } = render(<PokerItem />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerItem /> when render with default props.', () => {
        const { container } = render(<PokerItem />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', '100%');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('font-size', '1.8rem');
        expect(tree).toHaveStyleRule('border-radius', '10px');
        expect(tree).toHaveStyleRule('background-color', '#ffffff');
        expect(tree).toHaveStyleRule('cursor', 'pointer');
        expect(tree).toHaveStyleRule('overflow', 'hidden');
        expect(tree).toHaveStyleRule('position', 'relative');
    });

    it('test <PokerItem /> when render with borderColor props.', () => {
        const { container } = render(<PokerItem borderColor="#000" />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('background-color', '#000');
    });

    it('test <PokerItemMask /> snapshot.', () => {
        const { container } = render(<PokerItemMask />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerItemMask /> when render with default props.', () => {
        const { container } = render(<PokerItemMask />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('position', 'absolute');
        expect(tree).toHaveStyleRule('left', '0');
        expect(tree).toHaveStyleRule('top', '0');
        expect(tree).toHaveStyleRule('right', '0');
        expect(tree).toHaveStyleRule('bottom', '0');
        expect(tree).toHaveStyleRule('transition', 'all 200ms linear');
        expect(tree).toHaveStyleRule('z-index', '0');
        expect(tree).toHaveStyleRule('background', 'rgba(0,0,0,0)');
    });

    it('test <PokerItemMask /> when render with active props.', () => {
        const { container } = render(<PokerItemMask active />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('background', 'rgba(0,0,0,1)');
    });

    it('test <PokerItemInner /> snapshot.', () => {
        const { container } = render(<PokerItemInner />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerItemInner /> when render with default props.', () => {
        const { container } = render(<PokerItemInner />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '90%');
        expect(tree).toHaveStyleRule('height', '94%');
        expect(tree).toHaveStyleRule('border-radius', '10px');
        expect(tree).toHaveStyleRule('color', '#ffffff');
        expect(tree).toHaveStyleRule('background-color', '#5bb1d4');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'flex-end');
        expect(tree).toHaveStyleRule('align-items', 'flex-end');
        expect(tree).toHaveStyleRule('background-image', 'url(capoo-cat-logo-bg.png)');
        expect(tree).toHaveStyleRule('background-repeat', 'no-repeat');
        expect(tree).toHaveStyleRule('background-position', 'center center');
        expect(tree).toHaveStyleRule('background-size', 'cover');
        expect(tree).toHaveStyleRule('padding', '0 5px 5px 0');
        expect(tree).toHaveStyleRule('font-size', '1.5rem');
    });

    it('test <PokerItemInner /> when render with another props.', () => {
        const { container } = render(<PokerItemInner fontColor="#000" bgColor="#000" />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('color', '#000');
        expect(tree).toHaveStyleRule('background-color', '#000');
    });

    it('test <PokerSelectSafari /> snapshot.', () => {
        const { container } = render(<PokerSelectSafari />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerSelectSafari /> when render it.', () => {
        const { container } = render(<PokerSelectSafari />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('overflow', 'hidden');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('position', 'relative');
        expect(tree).toHaveStyleRule('border-radius', '25px');
    });

    it('test <PokerSelectMask /> snapshot.', () => {
        const { container } = render(<PokerSelectMask />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerSelectMask /> when render it.', () => {
        const { container } = render(<PokerSelectMask />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('z-index', '2');
        expect(tree).toHaveStyleRule('border-radius', '25px');
    });

    it('test <PokerSelectContainer /> snapshot.', () => {
        const { container } = render(<PokerSelectContainer />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerSelectContainer /> when render it.', () => {
        const { container } = render(<PokerSelectContainer />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('width', '100%');
        expect(tree).toHaveStyleRule('height', 'auto');
        expect(tree).toHaveStyleRule('content', '""', {
            modifier: ':after'
        });
        expect(tree).toHaveStyleRule('display', 'block', {
            modifier: ':after'
        });
        expect(tree).toHaveStyleRule('padding-bottom', '154%', {
            modifier: ':after'
        });
    });

    it('test <PokerSelectedItemFront /> snapshot.', () => {
        const { container } = render(<PokerSelectedItemFront />);

        expect(container.firstChild).toMatchSnapshot();
    });

    it('test <PokerSelectedItemFront /> when render with default props.', () => {
        const { container } = render(<PokerSelectedItemFront />);
        const tree = container.firstChild;

        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('border-radius', '25px');
        expect(tree).toHaveStyleRule('cursor', 'pointer');
        expect(tree).toHaveStyleRule('font-size', '5rem');
        expect(tree).toHaveStyleRule('color', '#5AC9E8');
        expect(tree).toHaveStyleRule('background-color', '#ffffff');
        expect(tree).toHaveStyleRule('background-size', 'contain');
        expect(tree).toHaveStyleRule('background-position', 'bottom');
        expect(tree).toHaveStyleRule('background-repeat', 'no-repeat');
        expect(tree).toHaveStyleRule('z-index', '1');

    });
});
