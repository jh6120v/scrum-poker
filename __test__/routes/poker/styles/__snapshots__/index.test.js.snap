// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Poker style. test <PokerContainer /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  -webkit-animation: iEMDfD;
  animation: iEMDfD;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  position: relative;
  z-index: 1;
}

@media (min-width:768px) {
  .c0 {
    width: 350px;
    margin: 0 auto;
  }
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerItem /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.8rem;
  border-radius: 10px;
  background-color: #ffffff;
  cursor: pointer;
  overflow: hidden;
  position: relative;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerItemInner /> snapshot. 1`] = `
.c0 {
  width: 90%;
  height: 94%;
  border-radius: 10px;
  color: #ffffff;
  background-color: #5bb1d4;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  background-image: url(capoo-cat-logo-bg.png);
  background-repeat: no-repeat;
  background-position: center center;
  background-size: cover;
  padding: 0 5px 5px 0;
  font-size: 1.5rem;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerItemMask /> snapshot. 1`] = `
.c0 {
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  -webkit-transition: all 200ms linear;
  transition: all 200ms linear;
  z-index: 0;
  background: rgba(0,0,0,0);
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerItemWrap /> snapshot. 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100px;
  max-width: 22%;
  margin: 6.5px 0;
}

.c0::after {
  content: '';
  display: block;
  padding-bottom: 140%;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerListWrap /> snapshot. 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  z-index: 1;
  padding: 0 20px;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerSelectContainer /> 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  -webkit-animation: iEMDfD;
  animation: iEMDfD;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: auto;
}

.c0:after {
  content: "";
  display: block;
  padding-bottom: 154%;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerSelectContainer /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  -webkit-animation: iEMDfD;
  animation: iEMDfD;
  -webkit-animation-fill-mode: forwards;
  animation-fill-mode: forwards;
  -webkit-animation-duration: 300ms;
  animation-duration: 300ms;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  position: relative;
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: auto;
}

.c0:after {
  content: "";
  display: block;
  padding-bottom: 154%;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerSelectMask /> snapshot. 1`] = `
.c0 {
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  -webkit-transition: all 200ms linear;
  transition: all 200ms linear;
  z-index: 0;
  background: rgba(0,0,0,0);
  z-index: 2;
  border-radius: 25px;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerSelectSafari /> snapshot. 1`] = `
.c0 {
  width: 100%;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 25px;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerSelectWrap /> snapshot. 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform: rotateY(180deg);
  -ms-transform: rotateY(180deg);
  transform: rotateY(180deg);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 1;
  padding: 0 20px;
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerSelectedItemFront /> snapshot. 1`] = `
.c0 {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 25px;
  cursor: pointer;
  font-size: 5rem;
  color: #5AC9E8;
  background-color: #ffffff;
  background-size: contain;
  background-position: bottom;
  background-repeat: no-repeat;
  z-index: 1;
}

.c0.img-1 {
  background-image: url(card-1.gif);
}

.c0.img-2 {
  background-image: url(card-2.gif);
}

.c0.img-3 {
  background-image: url(card-3.gif);
}

.c0.img-4 {
  background-image: url(card-4.gif);
}

.c0.img-5 {
  background-image: url(card-5.gif);
}

.c0.img-6 {
  background-image: url(card-6.gif);
}

.c0.img-7 {
  background-image: url(card-7.gif);
}

.c0.img-8 {
  background-image: url(card-8.gif);
}

.c0.img-9 {
  background-image: url(card-9.gif);
}

.c0.img-10 {
  background-image: url(card-10.gif);
}

.c0.img-11 {
  background-image: url(card-11.gif);
}

.c0.img-12 {
  background-image: url(card-12.gif);
}

.c0.img-13 {
  background-image: url(card-13.gif);
}

<div
  class="c0"
/>
`;

exports[`test Poker style. test <PokerWrap /> snapshot. 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  background-color: #aadff0;
  padding-top: 44px;
}

<div
  class="c0"
/>
`;
